internal/command/container.go

```diff
@@ -0,0 +1,398 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fyne-io/fyne-cross/internal/icon"
+	"github.com/fyne-io/fyne-cross/internal/log"
+	"github.com/fyne-io/fyne-cross/internal/volume"
+	"golang.org/x/sys/execabs"
+)
+
+const (
+	// fyneBin is the path of the fyne binary into the docker image
+	fyneBin = "/usr/local/bin/fyne"
+	// gowindresBin is the path of the gowindres binary into the docker image
+	gowindresBin = "/usr/local/bin/gowindres"
+)
+
+type Debugger interface {
+	Debug(v ...interface{})
+	Debugging() bool
+}
+
+type ContainerEngine interface {
+	CreateContainerImage(arch Architecture, OS string, image string) ContainerImage
+	Debugger
+}
+
+type baseEngine struct {
+	ContainerEngine
+
+	Env  map[string]string // Env is the list of custom env variable to set. Specified as "KEY=VALUE"
+	Tags []string          // Tags defines the tags to use
+
+	vol volume.Volume
+
+	debug bool
+}
+
+var _ Debugger = (*baseEngine)(nil)
+
+type ContainerImage interface {
+	Cmd(vol volume.Volume, opts Options, cmdArgs []string) *execabs.Cmd
+	Run(vol volume.Volume, opts Options, cmdArgs []string) error
+	Prepare() error
+	Finalize(srcFile string, packageName string) error
+
+	Architecture() Architecture
+	OS() string
+	ID() string
+	Target() string
+	Env(string) (string, bool)
+	SetEnv(string, string)
+	SetMount(string, string)
+	AppendTag(string)
+
+	Engine() ContainerEngine
```

Comment:
It seems strange that an `Image` has an `Engine` - is that the right way round for the relationship?

Reply:
It is necessary in many part of the code to find the Engine related to an Image being processed. Storing that relationship simplify the code greatly. Only relying on the Engine->Image relationship would likely result in adding the engine to every function call everywhere and reduce readability.

Reply:
I can see why it might need it internally - but why does it have to be exposed through the interface?